modules/core/28-wasm/handler.go

```diff
@@ -0,0 +1,20 @@
+package wasm
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/ibc-go/modules/core/28-wasm/keeper"
+	"github.com/cosmos/ibc-go/modules/core/28-wasm/types"
+)
+
+func HandleMsgPushNewWASMCode(ctx sdk.Context, k keeper.Keeper, msg *types.MsgPushNewWASMCode) (*types.MsgPushNewWASMCodeResponse, error) {
```

Comment:
It looks like anyone can create a new version of an existing wasm contract? Shouldn't there be some authentication in place?

Reply:
You're right. At this stage, anyone could push a new wasm-code and override someone else's client code. This is obviously bad, so I would imagine there must be ownership assigned to the account during the code upload.

I am not that familiar with governance, so any tips ideas on how this could be tackled are most welcome.

Reply:
I think [this comment](https://github.com/cosmos/ibc-go/pull/208#discussion_r654391280) explains the flow in more detail.

In short, anyone who is willing to pay transaction fees can upload the new light client code. The uploaded wasm-code is then identified by the `code_id` (sha256(code)) handle and must be passed by the client's call.

If someone pushes a new version of code, he'll get a new `code_id` handle, as if that was a new code